daffodil-runtime1/src/main/java/org/apache/daffodil/udf/UDFunctionService.java

```
@@ -0,0 +1,137 @@
+				}
+
+				fc.stream().forEach(f -> {
+					FunctionClassInfo fInfo = f.getAnnotation(FunctionClassInfo.class);
+					String ns = fInfo.namespace();
+					String fname = fInfo.name();
+					if (ns.isEmpty() || fname.isEmpty() || ns.matches("\\s+") || fname.matches("\\s+")) {
+						warnings.add(String.format("FunctionClass ignored: [%s]. From Provider [%s].%s%s", f.getName(), fcp.getClass().getName(),
+								ns.isEmpty() ? "\nAnnotation namespace field is empty or invalid." : "",
+								fname.isEmpty() ? "\nAnnotation name field is empty or invalid." : ""));
+					}
+
+					String key = String.join("_", fInfo.namespace(), fInfo.name());
+					if (functionProviderLookup.containsKey(key)) {
+						warnings.add(String.format(
+								"FunctionClass ignored: [%s].\nDuplicate name [%s] and namespace[%s] found.",
+								f.getName(), key.split("_")[0], key.split("_")[1]));
+					} else {
+						functionProviderLookup.put(key, fcp);
+						functionClasses.add(f);
+					}
+				});
+			});
+
+			if (functionClasses.isEmpty()) {
+				errors.add(String.format("No user defined functions found."
+						+ "\nCheck that UDF JARs are on classpath and that they are properly registerable by ServiceLoader."
+						+ "\nCurrent classpath locations:\n%s",
+						Arrays.asList(Misc.getClassPath()).stream().map(u -> u.toString())
+								.collect(Collectors.joining("\n"))));
+			}
+		} catch (ServiceConfigurationError | Exception e) {
+			e.printStackTrace();
+			errors.add(e.toString() + String.format("\nCurrent classpath locations:\n%s", Arrays
+					.asList(Misc.getClassPath()).stream().map(u -> u.toString()).collect(Collectors.joining("\n"))));
+		}
+	}
+
+	public Object lookupFunctionClass(String namespace, String name) {
+		Object funcObj = null;
+
+		try {
+			UDFunctionProvider fp = this.functionProviderLookup.get(String.join("_", namespace, name));
+			if (fp != null) {
+				funcObj = fp.lookupFunctionClass(namespace, name);
+			}
+		} catch (Exception e) {
```

You should avoid catch of broad things like Exception. 
In this case I think the right thing is remove the entire try/catch and just let the exception propagate.

---

Done